### ISSUE TITLE
service.restart salt-minion doesn't return True for some versions of minions

### ISSUE THREAD

**Issue by tjyang** [saltstack/salt#36106]
### Description of Issue/Question

 "salt minion01 service.restart salt-minion" doesn't return success even on the minion side the salt-mion process got a new pid after restart.

Version of salt-minions has problem: 2014.1.13, 2014.7.0, 2014.7.1 , 2015.8.3,  2016.3.3
Version of salt-minions has no problem: 2014.7.5, 2015.8.8.2, 2015.5.3, 2015.5.5, 2015.5.8, 2015.5.10, 2016.3.1, 2016.3.2, ubuntu 0.17.5, 2014.1.13
### Setup

(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
### Steps to Reproduce Issue

log on to minio01 to get current pid number
```
salt minion01 service.restart salt-minion
```
I got output of following,
```
minion01: 
  Minion did not return. [No response]
```
log on to minion01 and see the new pid number
### Versions Report

minion:
```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: unknown
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 3.2.4

System Versions:
           dist: centos 6.5 Final
        machine: x86_64
        release: 2.6.32-431.3.1.el6.x86_64
         system: Linux
        version: CentOS 6.5 Final
```
Master:
```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 0.8.6
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: 1.3.8
         Jinja2: 2.7.2
        libgit2: 0.21.0
        libnacl: 1.4.3
       M2Crypto: 0.21.1
           Mako: 0.8.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.2.3
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.21.4
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.28.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
@tjyang I am able to replicate this behavior on 2016.3.3 as well. Thanks we will need to get this fixed up and make sure its fixed for all releases.

**Comment by Ch3LL** [comment]
Also to note I am seeing this behavior in carbon. 

```
[root@localhost ~]# salt '*' service.restart salt-minion
10.0.2.15:
    Minion did not return. [No response]
```

**Comment by tjyang** [comment]
"service.restart salt-minion" works on unbunt 14.04 for version 2016.3.3

**Comment by tjyang** [comment]
@meggiebot , May I know why this bug won't be fixed in next release ?

**Comment by thatch45** [comment]
This bug is being continually worked on, and is very complicated. This is due to the many different init systems out there and the challenges they each represent. It is also noteworthy that we have had by far the most trouble with Upstart and Ubuntu.

We have introduced in the next feature release (Carbon) very tight integration with systemd to solve this issue and have spent a great deal of effort on it. Moving forward we will be focusing much more effort on systemd integration and much less on upstart due to the fact that systemd is replacing upstart on Ubuntu.

So the real answer is not that this will not be fixed, but rather, is that we feel that we have made significant progress here and continue to do so, but the recent changes for systemd were so extensive that we did not fee comfortable placing them in a point release.
